mono/metadata/icall-decl.h

```diff
@@ -198,6 +198,13 @@ ICALL_EXPORT gint32 ves_icall_System_IO_Compression_DeflateStreamNative_Flush (g
 ICALL_EXPORT gint32 ves_icall_System_IO_Compression_DeflateStreamNative_ReadZStream (gpointer stream, gpointer buffer, gint32 length);
 ICALL_EXPORT gint32 ves_icall_System_IO_Compression_DeflateStreamNative_WriteZStream (gpointer stream, gpointer buffer, gint32 length);
 #endif
+
+#if defined(TARGET_WASM)
+ICALL_EXPORT gpointer ves_icall_System_TimeZoneInfo_xamarin_timezone_get_data (MonoString* name, gint32 *size);
```

Comment:
The first and last should use handles, or perhaps other means.
For example, the first could use `fixed`.
I also suggest `ref to ref` but it isn't yet a widespread idiom.
Either way, the first and last are "wrong".

Reply:
Using handles requires access to the internal API's.  As tradeoff, think I am going to change this to use `mono_add_internal_call`.  This should wrap everything up nicely.

Reply:
I not sure how "nicely". This is rarely how things are done. But might be ok.
I suggest you avoid handles and their need using other means to solve the same problem:

1. For passing strings, use `fixed` and then `char*` and `int length`. There are a number of examples of this. For example the performance counters. It is quite easy and efficient and a blessed pattern.

2. Or similarly, you can convert to utf8 on the managed size. Pass a byte* then. I don't like it, but it occurs too. Probably requires extra managed/native roundtrips also.

3. Maybe something with p/invoke. I have no experience here but people use it.

3. For anything else non-MonoString, including the return value, pass a ref to local object. That is equivalent to a coop handle. See my PR removing handles from Monitor for example.
https://github.com/mono/mono/pull/17032
It was deemed ok but not actually approved. I don't know why.
Now it has conflicts.

And then you can say you are nohandles() and be coop-friendly.